jib-core/src/main/java/com/google/cloud/tools/jib/image/ImageReference.java

```diff
@@ -290,7 +290,8 @@ public String toString() {
 
     // Use tag if not the default tag.
     if (!DEFAULT_TAG.equals(tag)) {
-      referenceString.append(':').append(tag);
+      // Append with "@tag" instead of ":tag" if tag is a digest
+      referenceString.append(tag.matches(DescriptorDigest.DIGEST_REGEX) ? '@' : ':').append(tag);
```

Comment:
Should prob add this to `toStringWithTag` as well (perhaps with a `getTagString` method).

Reply:
Shouldn't need to, `getTagString` only does things differently if the tag is default (which it won't be if the tag is a digest).

Reply:
Ah right :+1: